src/doom/d_main.c

```diff
@@ -167,6 +172,8 @@ void D_ProcessEvents (void)
 
 // wipegamestate can be set to -1 to force a wipe on the next draw
 gamestate_t     wipegamestate = GS_DEMOSCREEN;
+snowflake_t     snowflakes[5000];
```

Comment:
Could you make this a function of SCREENHEIGHT and SCREENWIDTH, please? Something like `(SCREENHEIGHT * SCREENWIDTH) >> 6`. This will probably require dynamic allocation.

Reply:
Ok!